### PR TITLE
namespace: inital commit

### PR DESCRIPTION
**Pull request by sevki**
Signed-off-by: sevki <s@sevki.org>

### CODE REVIEW

File: namespace/builder.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &Builder{
+		dir:  wd,
+		open: open1,
+	}, nil
+}
+
+func newrunner(wd string, b OpenFunc) (*Builder, error) {
```
Comment (gmacd) [inline namespace/builder.go]: I guess we should go with the go naming conventions for consistency in the code - like newRunner or buildNs

File: namespace/builder.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	for _, c := range b.file {
+		var err error
+		args := []string{}
+		for _, arg := range c.args {
+			args = append(args, os.ExpandEnv(arg))
+		}
+
+		switch syzcall(c.syscall) {
+		case BIND:
+			err = ns.Bind(args[0], args[1], int(c.flag))
```
Comment (gmacd) [inline namespace/builder.go]: Has the number of args been validated by this time?

File: namespace/parser.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Sevki <s@sevki.org>. All rights reserved.
```
Comment (gmacd) [inline namespace/parser.go]: Best to replace with a non-personal license :)

**Review by gmacd — CHANGES_REQUESTED** [review]
Nice - good to see tests too!
Just a few superficial comments about naming conventions and licences, but the kind of thing we should get right in this first harvey-go code - this should set the example for others to follow.

File: namespace/parser.go (hunk 3 above)
Comment (sevki) [reply namespace/parser.go]: image

File: namespace/builder.go (hunk 2 above)
Comment (sevki) [reply namespace/builder.go]: yes.

**Review by gmacd — APPROVED** [review]

File: harvey_plan9.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
+// +build plan9
```
Comment (hugelgupf) [inline harvey_plan9.go:1]: this build tag is unnecessary and ineffective.

build tags need a fully empty new line after them to be effective.

however, naming the file _plan9.go already added this build tag, so it's unnecessary.

**Pull request merged**